go/test/endtoend/vtgate/system_schema_test.go

```diff
@@ -216,3 +220,43 @@ func TestMultipleSchemaPredicates(t *testing.T) {
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "specifying two different database in the query is not supported")
 }
+
+func TestSystemSchemaQueryWithUnion(t *testing.T) {
+	defer cluster.PanicHandler(t)
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	testcases := []struct {
+		predicates  []string
+		expectedKSs []string
+	}{{
+		predicates:  []string{"table_schema = 'ks'", "table_schema = 'performance_schema'"},
+		expectedKSs: []string{"vt_ks", "performance_schema"},
+	}, {
+		predicates:  []string{"table_schema in ('ks')", "table_schema in ('performance_schema')"},
+		expectedKSs: []string{"vt_ks", "performance_schema"},
+	}}
+	for _, tc := range testcases {
+		assertMultipleRowsAreReturned(t, conn, tc.predicates, tc.expectedKSs, len(tc.expectedKSs))
+	}
+}
+
+func assertMultipleRowsAreReturned(t *testing.T, conn *mysql.Conn, predicates []string, expectedKSs []string, expectedRows int) {
```

Comment:
this func name should be unionPredicatesAndAssertResults or something similar,

not some general assertMultiRowsAreReturned